modules/thunder_updater/src/Updater.php

```diff
@@ -0,0 +1,484 @@
+
+  /**
+   * Storage used for merged files.
+   *
+   * It should be access over: mergeFileStorage() method.
+   *
+   * @var StorageInterface
+   */
+  protected $mergeFileStorage;
+
+  /**
+   * Reverter that will use merged configuration files.
+   *
+   * @var ConfigReverter
+   */
+  protected $mergedConfigReverter;
+
+  /**
+   * {@inheritdoc}
+   */
+  public function __construct(
+    UpdaterConfigDiffer $config_diff,
+    ConfigListInterface $config_list,
+    ConfigRevertInterface $config_update,
+    ModuleHandlerInterface $module_handler,
+    EntityTypeManagerInterface $entity_manager,
+    StorageInterface $active_config_storage,
+    ConfigFactoryInterface $config_factory,
+    EventDispatcherInterface $dispatcher
+  ) {
+    $this->configDiffer = $config_diff;
+    $this->configList = $config_list;
+    $this->configReverter = $config_update;
+    $this->moduleHandler = $module_handler;
+    $this->entityManager = $entity_manager;
+    $this->configStorage = $active_config_storage;
+    $this->configFactory = $config_factory;
+    $this->eventDispatcher = $dispatcher;
+  }
+
+  /**
+   * Generate patch from changed configuration.
+   *
+   * It compares Base vs. Active configuration and creates patch with defined
+   * name in patch folder.
+   *
+   * @param string $module_name
+   *   Module name that will be used to generate patch for it.
+   * @param string $version_name
+   *   Suffix name for patch. Usually version number.
+   * @param string $patch_type
+   *   Defined how diff will be calculated.
+   *
+   * @return string|bool
+   *   Rendering generated patch file name or FALSE if patch is empty.
+   */
+  public function generatePatch($module_name, $version_name, $patch_type = self::PATCH_TYPE_REVERSE) {
+    $updatePatch = [];
+
+    $configurationLists = $this->configList->listConfig('module', $module_name);
+    $configNames = $this->getConfigNames($configurationLists[1]);
+
+    foreach ($configNames as $configName) {
+      $activeConfig = $this->getConfigFrom(
+        $this->configReverter->getFromActive($configName->getType(), $configName->getName())
+      );
+
+      $baseConfig = $this->getConfigFrom(
+        $this->configReverter->getFromExtension($configName->getType(), $configName->getName())
+      );
+
+      if (!$this->configDiffer->same($baseConfig, $activeConfig)) {
+        if ($patch_type == static::PATCH_TYPE_NORMAL) {
+          $updateDiff = $this->configDiffer->diff(
+            $baseConfig,
+            $activeConfig
+          );
+        }
+        else {
+          $updateDiff = $this->configDiffer->diff(
+            $activeConfig,
+            $baseConfig
+          );
+        }
+
+        $updatePatch[$configName->getFullName()] = $updateDiff->getEdits();
+      }
+    }
+
+    if (!empty($updatePatch)) {
+      $patchFilePath = $this->getPatchFileName($module_name, $version_name);
+      file_put_contents($patchFilePath, gzencode(serialize($updatePatch)));
```

Comment:
This fails, if the update directory has not been created before, can druapls file_save_data be used for this instead?